rust/src/delta_arrow.rs

```diff
@@ -42,7 +57,7 @@ impl TryFrom<&schema::SchemaTypeArray> for ArrowField {
 
     fn try_from(a: &schema::SchemaTypeArray) -> Result<Self, ArrowError> {
         Ok(ArrowField::new(
-            "element",
+            "item",
```

Comment:
This and the below changes are necessary to make them inline with the default field names in Rust and C++ Arrow, allowing easy round tripping between Arrow types and Delta Lake types.

Reply:
I think this change might cause issues when reading complex nested types. For example a map of arrays gives  
```
pyarrow.lib.ArrowNotImplementedError: Unsupported cast to map<string, list<item: string>> from map<string, list<element: string> ('map_arrays')>
```
It seems PyArrow cannot cast `element` -> `item` for some complex types.

Reply:
I created a test to reproduce this issue https://github.com/Tom-Newton/delta-rs/pull/13

Reply:
@wjones127 @Tom-Newton - is this something blocking us from mergi8n, or would this conflict somehow with #714?

Reply:
It's unrelated to #714. But it relates to making map types work correctly.

Reply:
great, this mean we can merge 714, right :)?

Reply:
Thanks for the info @Tom-Newton. I'll add that test case to this PR and make sure it is handled.

Reply:
I have reported an upstream issue: https://issues.apache.org/jira/browse/ARROW-17349

For now, I guess I'll see if I can align things with using `element`, since Spark seems to not allow customizing that field name.

Reply:
Tried two approaches and neither worked very well:

 1. First, I tried to change back to always convert Delta Array types to ListTypes with "element" as field name. With that change, we cannot roundtrip from `Arrow -> Delta -> Arrow` without having to re-map the field names, which I think will be annoying beyond just our test cases.
 2. Second, I tried changing so that whenever we read and write, we always use the "element" field name. But PyArrow errors when trying to write a table with a schema not matching the output schema, including field names. And if we try to cast we will run into exactly the same error you are getting now when reading.

Since this case isn't something we already support, I'm thinking the best course of action is to fix the issue upstream. Does that seem acceptable @Tom-Newton ?

Reply:
From my perspective this PR is a big improvement and does not introduce any regression, so I certainly think this is good to merge. However if this particular change was not needed it would have been even more helpful when it comes to making map types work.

I would like to understand why always using `element` doesn't work though.

Reply:
> I would like to understand why always using `element` doesn't work though.

First, there is the annoyance that is breaks roundtripping:

```python
pa_type = pa.list_(pa.int32()) # uses `item` as field name
delta_type = deltalake.schema.ArrayType.from_pyarrow(pa_type) # drops the field name
pa_type_back = delta_type.to_pyarrow() # adds `element` as field name
pa_type == pa_type_back # is False
```

But more importantly, it means that when *writing* to a Delta table, we need to make sure we use the field name "element" for list types. Otherwise when reading, we'll get the same error you were getting, but now complaining it can't cast from "item" to "element" instead of the other way around. That means when we write data, we have to cast any other field name to "element". But casting the field names is what's broken in the first place! So instead the writer would be broken in this edge case.

I guess ultimately it's a tradeoff: if we use "item", we break the reader in a certain edge case. If we use "element", we break the writer in that edge case *and* we break pyarrow type roundtripping. The latter seems slightly worse to me. Either way, we should have this fixed by the PyArrow 10.0.0, which should be released in October.

Reply:
Thanks for explaining 🙂 . So if I understand correctly the issue is because `write_deltalake` supports arbitrary pyarrow schemas so depending on whether the user's data uses `item` or `element` a cast may be needed. 

For the round tripping I guess you could make the opposite argument that `pa_type = pa.list_(pa.field("element", pa.int32()))` cannot be round tripped with the current implementation.